app/src/main/java/ingsw/group1/findmyphone/cryptography/SMSCipher.java

```diff
@@ -0,0 +1,135 @@
+package ingsw.group1.findmyphone.cryptography;
+
+import android.content.Context;
+
+import ingsw.group1.msglibrary.SMSMessage;
+import ingsw.group1.msglibrary.SMSPeer;
+
+/**
+ * Class used to cipher and decipher a SMSMessage.
+ *
+ * @author Pardeep Kumar
+ */
+public class SMSCipher implements MessageCipher<SMSMessage> {
+
+    private static final char PARSING_CHARACTER = '0';
+    private PasswordManager passwordManager;
+
+
+    public SMSCipher(Context context) {
```

Comment:
```suggestion
    //Important for the password to be final and .setPassword should be removed from class and interface, otherwise we lose track of which password we're using
    private final String password;
    public SMSCipher(String passwordToUse) {
        this.password = passwordToUse;
    }
```

Reply:
This way we make it clear that the Cipher has no clue where the password comes from, the password should be supplied from outside using PasswordManager instead of accessing it from here.